crates/interpreter/src/instructions/host.rs

```diff
@@ -366,6 +366,71 @@ pub fn static_call<H: Host, SPEC: Spec>(interpreter: &mut Interpreter<'_>, host:
     call_inner::<SPEC, H>(CallScheme::StaticCall, interpreter, host);
 }
 
+// EIP-3074
+pub fn auth<H: Host, SPEC: Spec>(interpreter: &mut Interpreter<'_>, host: &mut H) {
+    // Pop the `authority` off of the stack
+    pop_address!(interpreter, authority);
+    // Pop the signature offset and length off of the stack
+    pop!(interpreter, signature_offset, signature_len);
+
+    let signature_offset = as_usize_or_fail!(interpreter, signature_offset);
+    let signature_len = as_usize_or_fail!(interpreter, signature_len);
+
+    // Charge the fixed cost for the `AUTH` opcode
+    gas!(interpreter, gas::AUTH);
+
+    // Grab the memory region for the input data and charge for any memory expansion
+    shared_memory_resize!(interpreter, signature_offset, signature_len);
+    let mem_slice = interpreter
+        .shared_memory
+        .slice(signature_offset, signature_len);
+
+    // Pull the signature from the first 65 bytes of the memory region.
+    let signature = {
+        // Place the yParity value, located at the first byte, in the last byte.
+        let mut sig = [0u8; 65];
+        sig[0..64].copy_from_slice(&mem_slice[1..65]);
+        sig[64] = mem_slice[0];
+        sig
+    };
+
+    // If the memory region is longer than 65 bytes, pull the commit from the next [1, 32] bytes.
+    let commit = (mem_slice.len() > 65).then(|| {
```

Comment:
This is mostly to copy the behavior of Go's `copy` intrinsic, where it can perform partial reads. It would be nice to have some clarity here!

Immediately returning `0` makes sense, assuming the `commit` must be present (or at the very least, 0'd out explicitly).